### PR TITLE
Interactivity for neverhaveiever buttons

### PR DESCRIPTION
**Pull request by DanPlayz0**
This PR is meant to add interactivity to the neverhaveiever buttons. This is was a fun little side-project.

Here's a video of it working. Suggest any edits, if there's something you'd like.

https://github.com/user-attachments/assets/f6e87b88-d424-4062-a593-f72454353cdc

### CODE REVIEW

**Review by dpaulos6 — CHANGES_REQUESTED** [review]
If you change the theme while a custom message is visible, the message will be on top of the original message. The feature is a bit unreliable and needs improvement.


https://github.com/user-attachments/assets/c4bffca3-5d24-418d-8220-7cba1497dd3b

File: components/Embeds/NeverHaveIEverEmbed.tsx [hunk 1]
```diff
@@ -107,7 +120,93 @@ const NeverHaveIEverEmbed: FC<MainProps> = ({ replayedRounds }) => {
...
+          bot={profiles.wouldyou.bot}
+          verified={profiles.wouldyou.verified}
+        >
+          <p style={{ whiteSpace: "initial" }}>Click to see command</p>
+        </DiscordReply>
+          <DiscordEmbed
+            slot="embeds"
+            color={haveDone ? "#0091ff" : "#f00404"}
+            image={haveDone == null 
+              ? "https://quickchart.io/chart?c=%7Btype%3A%27outlabeledPie%27%2Cdata%3A%7Blabels%3A%5B%27Have%27%2C%27Have+not%27%5D%2Cdatasets%3A%5B%7BbackgroundColor%3A%5B%27%230091ff%27%2C%27%23f00404%27%5D%2Cdata%3A%5B1%2C1%5D%7D%5D%7D%2Coptions%3A%7Bplugins%3A%7Blegend%3Afalse%2Coutlabels%3A%7Btext%3A%27%25l+%25p%27%2Ccolor%3A%27white%27%2Cstretch%3A35%2Cfont%3A%7Bresizable%3Atrue%2CminSize%3A12%2CmaxSize%3A18%7D%7D%7D%7D%7D&w=750&h=750&ref=qc-js&bkg=%232F3136&f=png&v=2"
```
Comment (fb-sean) [inline components/Embeds/NeverHaveIEverEmbed.tsx]: I would actually copy the image instead of embed it by link to avoid third party requests from the user

Comment (dpaulos6) [reply components/Embeds/NeverHaveIEverEmbed.tsx]: I agree, it would probably load instantly and would be easier to handle.

Comment (fb-sean) [reply components/Embeds/NeverHaveIEverEmbed.tsx]: Didn't even think about that, only said that bcs of user privacy xD

**Comment by DanPlayz0** [comment]
> If you change the theme while a custom message is visible, the message will be on top of the original message. The feature is a bit unreliable and needs improvement.
> 
>  Recording.2024-07-30.123116.mp4

I've fixed this with the latest commit. Although not the best fix for it, it does now work...

File: components/Embeds/NeverHaveIEverEmbed.tsx (hunk 1 above)
Comment (DanPlayz0) [reply components/Embeds/NeverHaveIEverEmbed.tsx]: Implemented in [5b0bd6d](https://github.com/Would-You-Bot/website-app/pull/41/commits/5b0bd6db0ce87754a3c78cfd332297f286fd1032)

**Comment by mezotv** [comment]
> I've fixed this with the latest commit. Although not the best fix for it, it does now work...

Could you use the webp format for the images instead of png?

**Comment by dpaulos6** [comment]
My latest commit ([63ef884](https://github.com/Would-You-Bot/website-app/pull/41/commits/63ef88452ecb0f02b34d25fea55ec42e3a7bff2b)) addresses the file format change requested by @mezotv. Also improves the custom message behavior on theme switch, as it was currently removing the message after switching themes and now remains there in the same order.

**Comment by mezotv** [comment]
@dpaulos6 Would you mind attaching a video for me to check since I am currently on vacation?

**Comment by DanPlayz0** [comment]
Related to [10c91ae](https://github.com/Would-You-Bot/website-app/pull/41/commits/10c91ae43edd7d4a48b8548cbfc93a288238ee98), might as well delete my `EphemeralRow` component...

**Comment by dpaulos6** [comment]
> Related to [10c91ae](https://github.com/Would-You-Bot/website-app/pull/41/commits/10c91ae43edd7d4a48b8548cbfc93a288238ee98), might as well delete my `EphemeralRow` component...

Oh yea, completely forgot.

**Comment by dpaulos6** [comment]
As soon as I know the fix to the discord buttons without manually adding flex, this PR will be good to go afaik.
Anything missing or any cool stuff that could be added, just add a comment here.

**Comment by DanPlayz0** [comment]
Should the `New Question` button also be made interactive?

**Comment by mezotv** [comment]
> Should the `New Question` button also be made interactive?

If possible yes while also clearing the voted states

**Comment by DanPlayz0** [comment]
Here's the current version of these commits. Although there seems to be a bug with light theme and the ephemeral message text color and a random curly bracket where it says "Click to see command`}`".

https://github.com/user-attachments/assets/9ca4d041-3f2d-41dc-b9e0-d6fc421d8b2d

**Comment by fb-sean** [comment]
Can someone somehow remove me from this PR I don't want to receive notifies for every commit change 😭

//Edit: Found the unsubscribe button...

**Comment by dpaulos6** [comment]
> Here's the current version of these commits. Although there seems to be a bug with light theme and the ephemeral message text color and a random curly bracket where it says "Click to see command`}`".
> 
>  2024-08-02.00-53-31.mp4

I just fixed the color bug on the light theme, for some reason some components are not applying white theme.

Related to the `}` char at the end of the message, it's a very weird bug and it's not present in any part of the code. Can you read the docs on [skyra/discord-components - migrating to v4](https://github.com/skyra-project/discord-components/blob/main/packages/core/UPGRADING-TO-V4.md) and see if you can find anything that needs changing? I recently updated the version of it to use the ephemeral hook, so can you confirm for me everything is correct with the docs on v4?

**Comment by DanPlayz0** [comment]
> > Here's the current version of these commits. Although there seems to be a bug with light theme and the ephemeral message text color and a random curly bracket where it says "Click to see command`}`".
> > 2024-08-02.00-53-31.mp4
> 
> I just fixed the color bug on the light theme, for some reason some components are not applying white theme.
> 
> Related to the `}` char at the end of the message, it's a very weird bug and it's not present in any part of the code. Can you read the docs on [skyra/discord-components - migrating to v4](https://github.com/skyra-project/discord-components/blob/main/packages/core/UPGRADING-TO-V4.md?rgh-link-date=2024-08-02T13%3A29%3A17Z) and see if you can find anything that needs changing? I recently updated the version of it to use the ephemeral hook, so can you confirm for me everything is correct with the docs on v4?

The rouge `}` at the end of the message, after enabling `command={true}`, is now more obvious that is within the source code of the package. I have submitted a PR (https://github.com/skyra-project/discord-components/pull/437) over there to fix this. Might take a bit to get it approved.

Here's the another screenshot (with the command set to true) that you can see the curly brace,
image

**Comment by mezotv** [comment]
> The rouge `}` at the end of the message, after enabling `command={true}`, is now more obvious that is within the source code of the package. I have submitted a PR (https://github.com/skyra-project/discord-components/pull/437) over there to fix this. Might take a bit to get it approved.

This was just merged! Just run pnpm i @skyra/discord-components-react@4.0.0-alpha.26

**Comment by dpaulos6** [comment]
> > The rouge `}` at the end of the message, after enabling `command={true}`, is now more obvious that is within the source code of the package. I have submitted a PR ([skyra-project/discord-components#437](https://github.com/skyra-project/discord-components/pull/437)) over there to fix this. Might take a bit to get it approved.
> 
> This was just merged! Just run pnpm i @skyra/discord-components-react@4.0.0-alpha.26

Done on [65b3866](https://github.com/Would-You-Bot/website-app/pull/41/commits/65b3866ea329639e739e38f1e1ae78f28158a312).
Also when using pnpm its recommended to do `pnpm add` when installing modules.

**Review by dpaulos6 — APPROVED** [review]

**Review by dpaulos6 — CHANGES_REQUESTED** [review]
@DanPlayz0 can you please check the button methods on skyra docs about the emojis? right now we have a custom span with flex in order for it to work, but supposedly `discord-button` has `emoji` and `emojiName` attributes.

If you can take a look at it, i'd really appreciate. If you are unable i can worry about it tomorrow.

**Comment by DanPlayz0** [comment]
> @DanPlayz0 can you please check the button methods on skyra docs about the emojis? right now we have a custom span with flex in order for it to work, but supposedly `discord-button` has `emoji` and `emojiName` attributes.
> 
> If you can take a look at it, i'd really appreciate. If you are unable i can worry about it tomorrow.

I've done a bit of testing with the properties you mentioned.

The `discord-button#emoji` property takes in a url, not an emoji or jsx component. The `discord-button#emojiName` property is just the alternative name for the img element when it doesn't load. It seems we will have to convert all of our hardcoded/inline svgs into files within the public directory.

I am a bit un-experienced with SVGs so when I moved the code for an svg out and into a file, it turned out to look like this.
image
Compared to the original (inline/hardcoded) 
image
There seems to be some sort of `currentColor` variable set when it's inline.

**Comment by dpaulos6** [comment]
> > @DanPlayz0 can you please check the button methods on skyra docs about the emojis? right now we have a custom span with flex in order for it to work, but supposedly `discord-button` has `emoji` and `emojiName` attributes.
> 
> > 
> 
> > If you can take a look at it, i'd really appreciate. If you are unable i can worry about it tomorrow.
> 
> 
> 
> I've done a bit of testing with the properties you mentioned.
> 
> 
> 
> The `discord-button#emoji` property takes in a url, not an emoji or jsx component. The `discord-button#emojiName` property is just the alternative name for the img element when it doesn't load. It seems we will have to convert all of our hardcoded/inline svgs into files within the public directory.
> 
> 
> 
> I am a bit un-experienced with SVGs so when I moved the code for an svg out and into a file, it turned out to look like this.
> 
> image
> 
> Compared to the original (inline/hardcoded) 
> 
> image
> 
> There seems to be some sort of `currentColor` variable set when it's inline.
> 
> 
> 
> 

i can fix that, push it with the implementation and i'll fix the colors.

**Comment by DanPlayz0** [comment]
> i can fix that, push it with the implementation and i'll fix the colors.

Just pushed my implementation with commit [672d051](https://github.com/Would-You-Bot/website-app/pull/41/commits/672d051133bcae12d9102d1d6444a7fa0c0614e0)

**Comment by dpaulos6** [comment]
@DanPlayz0 In this case, the `external.svg` was using `fill="currentColor"`, which is a dynamic property that changes the SVG's vectors to a color set in a CSS class. That allows us to have more accessibility to SVG's within a project, and for example, if you were exporting the SVG as an icon, you could add `className="text-green-500"` to change the SVG's color.

Hope that helps you understand a bit more about SVG's.

**Comment by DanPlayz0** [comment]
> @DanPlayz0 In this case, the `external.svg` was using `fill="currentColor"`, which is a dynamic property that changes the SVG's vectors to a color set in a CSS class. That allows us to have more accessibility to SVG's within a project, and for example, if you were exporting the SVG as an icon, you could add `className="text-green-500"` to change the SVG's color.
> 
> Hope that helps you understand a bit more about SVG's.

I had a feeling it was related to `currentColor`, but I was expecting to have to set a variable to make it work, rather than just changing it to a hex code. Thank for that knowledge!

**Review by dpaulos6 — APPROVED** [review]
This PR is completed in my opinion. Unless there's features in mind to be implemented further, this is ready to be merged.

**Pull request merged**